internal/http/logs.go

```diff
@@ -87,48 +92,139 @@ func (h logsHandler) files(w http.ResponseWriter, r *http.Request) {
+	uhdbotInviteRegex      = regexp.MustCompile(`(UHDBot invite)(\s+)([a-zA-Z0-9]+)`)
+	endorInviteRegex       = regexp.MustCompile(`(ENDOR !invite(\s+)\w+).([a-zA-Z0-9]+)`)
+	vertigoInviteRegex     = regexp.MustCompile(`(Vertigo ENTER #GGn-Announce\s+)(\w+).([a-zA-Z0-9]+)`)
+	immortalInviteRegex    = regexp.MustCompile(`(immortal invite(\s+)\w+).([a-zA-Z0-9]+)`)
+	muffitInviteRegex      = regexp.MustCompile(`(Muffit bot #nbl-announce\s+\w+)(\s+[a-zA-Z0-9]+)`)
+	hermesInviteRegex      = regexp.MustCompile(`(hermes enter #announce\s+\w+).([a-zA-Z0-9]+)`)
+	hummingbirdInviteRegex = regexp.MustCompile(`(Hummingbird ENTER\s+\w+).([a-zA-Z0-9]+)(\s+#ptp-announce-dev)`)
+	droneInviteRegex       = regexp.MustCompile(`(Drone enter #red-announce\s+\w+).([a-zA-Z0-9]+)`)
+	revottInviteRegex      = regexp.MustCompile(`(RevoTT !invite\s+\w+).([a-zA-Z0-9]+)`)
+	scenehdInviteRegex     = regexp.MustCompile(`(SceneHD..invite).([a-zA-Z0-9]+)(\s+#announce)`)
+	ericaInviteRegex       = regexp.MustCompile(`(erica letmeinannounce\s+\w+).([a-zA-Z0-9]+)`)
+	cerberusInviteRegex    = regexp.MustCompile(`(Cerberus identify\s+\w+).([a-zA-Z0-9]+)`)
 )
 
-func SanitizeLogFile(filePath string) (string, error) {
-	data, err := ioutil.ReadFile(filePath)
+// SanitizeLogFile reads a log file line by line and sanitizes each line using regular expressions.
+// It uses a worker pool to process multiple lines concurrently.
+func SanitizeLogFile(filePath string) (io.Reader, error) {
+	inFile, err := os.Open(filePath)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
+	defer inFile.Close()
 
-	sanitizedData := string(data)
+	reader := bufio.NewReader(inFile)
 
-	// torrent_pass, passkey, authkey, secret_key, apikey, rsskey
-	sanitizedData = keyValueRegex.ReplaceAllString(sanitizedData, "${1}=REDACTED")
-	sanitizedData = combinedRegex.ReplaceAllString(sanitizedData, "${1}REDACTED")
+	sanitizedContent := &bytes.Buffer{}
 
-	// irc related
-	sanitizedData = inviteRegex.ReplaceAllString(sanitizedData, "${1}REDACTED")
-	sanitizedData = nickservRegex.ReplaceAllString(sanitizedData, "${1}REDACTED")
-	sanitizedData = saslRegex.ReplaceAllString(sanitizedData, "${1}REDACTED")
-
-	tmpFile, err := ioutil.TempFile("", "sanitized-log-*.log")
-	if err != nil {
-		return "", err
+	// Define the number of worker goroutines
+	numCPUs := runtime.NumCPU()
+	numWorkers := numCPUs
+	if numCPUs <= 2 {
+		numWorkers = 1
 	}
 
-	_, err = tmpFile.WriteString(sanitizedData)
-	if err != nil {
-		tmpFile.Close()
-		return "", err
+	// Mutex to ensure only one worker reads a line and writes the sanitized line at a time
+	fileMutex := sync.Mutex{}
+
+	// Create a WaitGroup to wait for all workers to finish
+	wg := sync.WaitGroup{}
+
+	// Start the worker goroutines
+	for i := 0; i < numWorkers; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			for {
+				// Read the next line from the file
+				fileMutex.Lock()
+				line, err := reader.ReadString('\n')
```

Comment:
I think this may not be doing what you would expect. Writes will still not be ordered correctly. Going back to the single thread is cleaner and may improve overall times on these machines. The alternative is to create a channel array for your workers by idx, then receive on them by ranging over them until they're all closed.